prow/getting_started_kind_minikube.md

```diff
@@ -0,0 +1,109 @@
+This document is targeting developers, testers, and all the other interested parties who would like to learn or test prow, but do not have resources other than a localhost workstation. It describes a simplified minikube/kind deployment of crucial prow components:
+
+* [`crier`](prow/cmd/crier)
+* [`deck`](prow/cmd/deck)
+* [`hook`](prow/cmd/hook)
+* [`horologium`](prow/cmd/horologium) 
+* [`prow-controller-manager`](prow/cmd/prow-controller-manager) 
+* [`sinker`](prow/cmd/sinker) 
+* [`tide`](prow/cmd/tide)
+* [`status-reconciler`](prow/cmd/status-reconciler)
+* [`ghproxy`](ghproxy)
+
+Additionally, besides prow components, minio is set up as blob storage for logs.
+
+More information about each component function can be found [here](prow/cmd/). Other components can be easily added as well, depending on user needs.
+
+## Prerequisites
+
+* Create a new GitHub repository or choose an existing one that will be connected to your localhost prow.
+* Please follow [**GitHub App**](prow/getting_started_deploy.md) section from **Deploying Prow** tutorial. During configuration please be aware, that fields like ``Homepage URL`` or ``Callback URL`` don't matter that much in the test deployment. You can skip them, and if not possible, an invalid URL also should work. The most important field for this tutorial ``Webhook URL`` should be filled by following the instruction from the next paragraph.
+* Install configured application into your project (``Settings -> Developer settings -> GitHub Apps -> YourAppName -> Edit -> Install App``)
+* Install docker and kind or minikube. 
+
+## Filling Webhook URL
+
+The deployment will work on your localhost machine, but it needs to connect to the GitHub application configured in the previous section. There are several methods to do this with limited access to the public IP. The simplest would be to use dedicated services webhook services. A good example of such a service could be [smee.io](https://smee.io/). Such services should be used only for testing and development purposes. Please visit [smee.io/new](https://smee.io/new) to generate a new channel. Follow the instruction to install the tool (using ``npm``). Paste the generated link in the ``Webhook URL`` config field in your application configuration.
+
+The ``Webhook URL`` should be configured with a secret. To generate a secret please run in the console:
+```
+$ openssl rand -hex 20
+```
+Save the result locally, and paste it in the field ``Webhook secret (optional)`` during GitHub App configuration.
+
+## Deployment on kind or minikube
+
+### kind preparation
+Start a new kind cluster using:
+```
+kind create cluster
+```
+After a proper startup, you need IP address assigned to kind. To get IP assigned to kind you can execute the following command:
+```
+$ kubectl --context kind-kind  describe service kubernetes
+```
+IP is listed in `Endpoints:` section. If there is any port, ignore it. Finally, execute `smee` command. Taking `172.18.0.2` as an example IP run:
```

Comment:
Maybe there is a simpler method of getting kind IP address assigned by docker? I am rather minikube user and I think there is no equivalent of `minikube ip`...